src/query_builder/knex_base.ts

```diff
@@ -116,42 +117,35 @@ export default class KnexQueryBuilder implements IQueryBuilder<Knex> {
     // [string, string, string | number | null]
     // tslint:disable-next-line: cyclomatic-complexity
     private filterArgs(filter: IFilter) {
-        const transformedFilter = this.filterTransformer(filter);
-
-        const filterIsNullComparison =
-            transformedFilter.value === null ||
-            (typeof transformedFilter.value === 'string' &&
-                transformedFilter.value.toLowerCase() === 'null');
-
-        if (
-            this.useSuggestedValueLiteralTransforms &&
-            filterIsNullComparison &&
-            transformedFilter.operator.toLowerCase() === '='
-        ) {
-            return [
-                (builder: QueryBuilder) => {
-                    builder.whereNull(this.computeFilterField(transformedFilter.field));
-                }
-            ];
-        }
+        const {field, operator, value} = this.filterTransformer(filter);
+
+        if (this.useSuggestedValueLiteralTransforms) {
```

Comment:
This is enabled by default, and causes us to attempt to cast a provided string filter value to a native type, e.g.

```
'true' => true
'false' => false
'FALSE' => false
'1.1242' => 1.1242
'4' => 4
'A regular old string' => 'A regular old string'
```